File: a121(prime_again)/C++/prime_again.cpp
#include <iostream>
#include <cmath>

bool is_prime(int number) {
  if(number == 2 || number == 3) {
    return true;
  } else if(number % 6 != 1 && number % 6 != 5) {
      return false;
  } else {
    int ceil = sqrt(number);
    for(int i = 5; i <= ceil; i += 6) {
      if(number % i == 0 || number % (i+2) == 0) {
        return false;
      }
    }
  }
  return number != 1; // 1 is not a prime number
}


int main(void) {
  int start, end;
  while (std::cin >> start >> end) {
    int count = 0;
    // start with 1 and 2 need to be handled
    if (start == 2 || (start == 1 && end != 1) ) {
      count = 1;
      for (int i = 3; i <= end; i += 2) {
        if (is_prime(i)) count++;
      }
    } else if (start % 2 == 0) {
      for (int i = start + 1; i <= end; i += 2) {
        if (is_prime(i)) count++;
      }
    } else {
      for (int i = start; i <= end; i += 2) {
        if (is_prime(i)) count++;
      }
    }
    std::cout << count << '\n';
  }

  return 0;
}

File: a121(prime_again)/Python/prime_again.py

def is_prime(number: int) -> bool:
    if number == 2 or number == 3:
        return True
    elif number % 6 != 1 and number % 6 != 5:
        return False;
    else:
        ceil = int(number ** 0.5);
    for i in range(5, ceil + 1, 6):
        if number % i == 0 or number % (i+2) == 0:
            return False
    # 1 is not a prime number
    return number != 1


while True:
    try:
        start, end = map(int, input().split())
        count = 0;
        # start with 1 and 2 need to be handled
        if start == 2 or (start == 1 and end != 1):
            count = 1;
            for i in range(3, end + 1, 2):
                if is_prime(i): count += 1
        elif start % 2 == 0:
            for i in range(start + 1, end + 1, 2):
                if is_prime(i): count += 1
        else:
            for i in range(start, end + 1, 2):
                if is_prime(i): count += 1
        print(count)
        
    except:
        break
